additionalInstallationScripts/installWazuhAgent.sh

```diff
@@ -21,7 +21,24 @@ yum install wazuh-agent -y -q -e 0
 # register agent
 MANAGER_IP="172.16.0.21"
 
-until /var/ossec/bin/agent-auth -m $MANAGER_IP -A "linuxVictim"; do
+# get host IP
+hostIP=$(hostname -i)
+agentName="wazuhAgent"
+
+if [ "$hostIP" == "172.16.0.22" ]
+then
+  agentName="linuxVictim"
+elif [ "$hostIP" == "172.16.0.20" ]
+then
+  agentName="redTeam"
+elif [ "$hostIP" == "172.16.0.21" ]
+then
+  agentName="wazuhServer"
+else
+  agentName="bastion"
+fi
+  
+until /var/ossec/bin/agent-auth -m $MANAGER_IP -A $agentName; do
```

Comment:
According to the [`agent-auth` documentation](https://documentation.wazuh.com/current/user-manual/reference/tools/agent-auth.html), `hostname` will be used as agent name if no other is specified under `-A` option. I think it's more elegant and maintainable to configure instances hostname in the template and remove `-A` option from here 😄 .

As a possible way to do it, I found this [stackoverflow thread](https://stackoverflow.com/questions/40915572/setting-aws-ec2-instance-hostname-using-cloudformation-template).

Reply:
made this change added code to set the instance hostname on first boot in commit https://github.com/sonofagl1tch/AWSDetonationLab/pull/47/commits/c7596c513f3d0dd32239747d9a29a13f35345047

Reply:
Fixed in 73080b91092ec8c55c9a8a14d7da326910eabc1e
```shellsession
# /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuhServer (server), IP: 127.0.0.1, Active/Local
   ID: 001, Name: linuxVictim, IP: 172.16.0.22, Active
   ID: 002, Name: windowsVictim, IP: 172.16.0.23, Active
   ID: 003, Name: bastion, IP: 172.16.0.11, Active
   ID: 004, Name: redTeam, IP: 172.16.0.20, Active
```